### PR TITLE
LPD-24715 Add a testSelection() method to CategoryFacetTest

### PR DESCRIPTION
**Pull request by felipelorenz**
https://liferay.atlassian.net/browse/LPD-24715
Continued from here: https://github.com/liferay-search/liferay-portal/pull/1331

### CODE REVIEW

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 1]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
+			Collections.singletonMap(
+				_getAssetVocabularyCategoryId(_assetCategory), 1));
+	}
+
+	@Test
+	public void testSelection() throws Exception {
+		addCategory(RandomTestUtil.randomString());
 
-		AssetCategory assetCategory = addCategory(title);
+		addJournalArticle(_group, _assetCategory.getTitleCurrentValue());
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:135]: you are passing information's from global variables, can you remove the parameters from this method?

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 2]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
 
-		long categoryId = assetCategory.getCategoryId();
+		long categoryId = _assetCategory.getCategoryId();
 
 		addUser(_group, categoryId);
 
-		SearchContext searchContext = getSearchContext(
-			assetCategory.getTitleCurrentValue());
+		SearchContext searchContext = _getSearchContext(
+			_assetCategory, categoryId);
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:142]: same here

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 3]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
+	public void testSelection() throws Exception {
+		addCategory(RandomTestUtil.randomString());
 
-		AssetCategory assetCategory = addCategory(title);
+		addJournalArticle(_group, _assetCategory.getTitleCurrentValue());
 
-		long categoryId = assetCategory.getCategoryId();
+		long categoryId = _assetCategory.getCategoryId();
 
 		addUser(_group, categoryId);
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:139]: same here

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 4]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
-			assetCategory.getTitleCurrentValue());
+		SearchContext searchContext = _getSearchContext(
+			_assetCategory, categoryId);
 
-		searchContext.setCategoryIds(new long[] {categoryId});
-		searchContext.setGroupIds(new long[] {_group.getGroupId()});
+		Facet facet = _categoryFacetFactory.newInstance(searchContext);
 
-		Facet facet = categoryFacetFactory.newInstance(searchContext);
+		facet.select(_getAssetVocabularyCategoryId(_assetCategory));
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:146]: same here

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 5]
```diff
@@ -152,27 +207,48 @@ protected void addUser(Group group, long... categoryIds) throws Exception {
 		UserTestUtil.updateUser(_user, serviceContext);
 	}
 
-	@Inject
-	protected AssetCategoryLocalService assetCategoryLocalService;
-
-	@Inject
-	protected AssetVocabularyLocalService assetVocabularyLocalService;
+	protected void assertEntryClassNames(
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:210]: private

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 6]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
+						new JournalArticleTitle() {
+							{
+								put(LocaleUtil.US, title);
+							}
+						});
+				}
+			});
 	}
 
 	protected void addUser(Group group, long... categoryIds) throws Exception {
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:198]: private

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 7]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
+	protected void addCategory(String title) throws Exception {
+		_assetCategory = _assetCategoryLocalService.addCategory(
 			TestPropsValues.getUserId(), _group.getGroupId(), title,
 			_assetVocabulary.getVocabularyId(),
 			ServiceContextTestUtil.getServiceContext(
 				_group.getGroupId(), TestPropsValues.getUserId()));
+	}
 
-		_assetCategory = assetCategory;
+	protected void addJournalArticle(Group group, String title)
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:170]: private

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 8]
```diff
@@ -101,43 +109,90 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
 		FacetsAssert.assertFrequencies(
-			facet.getFieldName(), searchContext, hits, frequencies);
+			facet.getFieldName(), searchContext, hits,
+			Collections.singletonMap(
+				_getAssetVocabularyCategoryId(_assetCategory), 1));
 	}
 
-	protected AssetCategory addCategory(String title) throws Exception {
-		AssetCategory assetCategory = assetCategoryLocalService.addCategory(
+	protected void addCategory(String title) throws Exception {
```
Comment (gustavolimav) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:162]: private

**Review by gustavolimav — CHANGES_REQUESTED** [review]

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 9]
```diff
@@ -101,78 +105,137 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
+									_assetCategory.getTitleCurrentValue());
+							}
+						});
+				}
+			});
 	}
 
-	protected void addUser(Group group, long... categoryIds) throws Exception {
-		_user = UserTestUtil.addUser(group.getGroupId());
+	private void _addUser() throws Exception {
```
Comment (AlmirFe) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:186]: All tests in this class are using this function to add a user. You could reduce the complexity of the code by adding the user in the setUp function. See https://github.com/AlmirFe/liferay-portal/commit/b82762fc9de2f4cae10c353924a26a4f12b1b2d1

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 10]
```diff
@@ -101,78 +105,137 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
 			TestPropsValues.getUserId(), _group.getGroupId(), title,
 			_assetVocabulary.getVocabularyId(),
 			ServiceContextTestUtil.getServiceContext(
 				_group.getGroupId(), TestPropsValues.getUserId()));
+	}
 
-		_assetCategory = assetCategory;
-
-		return assetCategory;
+	private void _addJournalArticle() throws Exception {
```
Comment (AlmirFe) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:158]: The "throws exception" part is unnecessary

**Review by AlmirFe — CHANGES_REQUESTED** [review]
Hi @felipelorenz, I requested some simple changes to the CategoryFacetTest class. Let me know if you have any questions

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java [hunk 11]
```diff
@@ -101,78 +105,137 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
...
+		journalArticleSearchFixture.addArticle(
+			new JournalArticleBlueprint() {
```
Comment (AlmirFe) [inline modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:160]: Use [JournalArticleBlueprintBuilder](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-test-util/src/main/java/com/liferay/journal/test/util/search/JournalArticleBlueprintBuilder.java). See https://github.com/AlmirFe/liferay-portal/commit/2cbd69c19ac68c9912d5fa11408c7ad3a6b4bac1

**Review by AlmirFe — CHANGES_REQUESTED** [review]

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

File: modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java (hunk 11 above)
Comment (felipelorenz) [reply modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java:167]: Hi @AlmirFe, for this change I think it's more a matter of preference of each pattern, I'll keep it the way it is to follow the standard, I see that the facets that use this method, use this way, but we can create a ticket to discuss which would be better and standardize.

**Review by AlmirFe — APPROVED** [review]

**Comment by gustavolimav** [comment]
ci:forward

**Pull request closed without merging**